core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -190,25 +190,52 @@ oppia.directive('questionsList', [
               controller: [
                 '$scope', '$uibModalInstance',
                 function($scope, $uibModalInstance) {
-                  $scope.selectedSkillIds = [];
+                  var fillSkillDifficulties = function(length) {
+                    $scope.selectedSkillDifficulties = [];
+                    for (var i = 0; i < length; i++) {
+                      $scope.selectedSkillDifficulties.push('0.3');
+                    }
+                  };
+                  $scope.selectedSkillIds = linkedSkillIds;
+                  $scope.selectedSkillDescriptions = [];
+                  fillSkillDifficulties(ctrl.newQuestionSkillIds.length);
+                  $scope.selectDifficulty = selectDifficulty;
                   $scope.skillSummaries = allSkillSummaries;
-                  $scope.skillSummaries.forEach(function(summary) {
-                    summary.isSelected = false;
-                  });
+                  if ($scope.skillSummaries) {
+                    $scope.skillSummaries.forEach(function(summary) {
+                      summary.isSelected = false;
+                    });
+                  }
 
                   $scope.selectOrDeselectSkill = function(skillId, index) {
                     if (!$scope.skillSummaries[index].isSelected) {
                       $scope.selectedSkillIds.push(skillId);
+                      $scope.selectedSkillDescriptions.push(
+                        $scope.skillSummaries[index].getDescription());
                       $scope.skillSummaries[index].isSelected = true;
                     } else {
                       var idIndex = $scope.selectedSkillIds.indexOf(skillId);
                       $scope.selectedSkillIds.splice(idIndex, 1);
+                      $scope.selectedSkillDescriptions.splice(idIndex, 1);
                       $scope.skillSummaries[index].isSelected = false;
                     }
                   };
 
+                  $scope.back = function() {
```

Comment:
Please use a name that starts with a verb, like $scope.goToPreviousPage or $scope.closeModal or $scope.cancelModal or similar.

Ditto for next() and done() below.

Reply:
Done